crates/storage/provider/src/providers/historical.rs

```diff
@@ -64,6 +64,24 @@ impl<'a, 'b, TX: DbTx<'a>> AccountProvider for HistoricalStateProviderRef<'a, 'b
             Ok(self.tx.get::<tables::PlainAccountState>(address)?)
         }
     }
+
+    /// Get account code by its address
+    fn account_code(&self, addr: Address) -> Result<Option<Bytes>> {
+        // Get basic account information
+        let acc = match self.basic_account(addr)? {
+            Some(acc) => acc,
+            None => return Ok(None),
+        };
+        // Get code hash
+        let code_hash = acc.bytecode_hash;
+        // Get the code from the code hash
```

Comment:
this needs to check if the `bytecode_hash` is `KECCAK_EMPTY` in which case the account does not have any code and can return `None`